woocommerce/woocommerce
# Improve rendering of order list table on mobile

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR makes some changes to the responsiveness enhancements added in #47684, which worked pretty well for smaller screen sizes, except particularly on certain mobile devices.

While there are other ways to get to order details (by expanding the details, clicking the "Preview" icon or simply tapping the entire row), the most common way to do this is by clicking the link with the order number and buyer name, and so this PR prioritizes always showing this target.

The following table shows a comparison of how the list table looks at various screen sizes both in 9.1.2 (before) and this PR (after). In particular, for smaller sizes, in 9.1.2 the link is barely visible or entirely hidden, while it remains always visible now.


| Screen Width | Before (9.1.2) | After |
|--------|--------|--------|
| 390px (iPhone 12/13) | <img src="https://github.com/user-attachments/assets/702c5597-833f-4dd4-a5ed-30dc52ad9dcf" width="150" /> | <img src="https://github.com/user-attachments/assets/9f448804-f15b-4ddc-90fa-48ae5fb05920" width="150" /> |
| 450px | <img src="https://github.com/user-attachments/assets/90f7604a-cd37-4c77-8a07-8b1015276d7a" width="150" /> | <img src="https://github.com/user-attachments/assets/8f8118fa-8bfd-4e2f-a807-e7e9373f9267" width="150" /> |
| 500px | <img src="https://github.com/user-attachments/assets/b338c57e-b87a-4510-b7d1-c467b7bba118" width="150" /> | <img src="https://github.com/user-attachments/assets/ebeffb6f-3240-4d78-b009-86fc73cd976a" width="150" /> |


Closes #49522.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have a few orders on your site. [Smooth Generator](https://github.com/woocommerce/wc-smooth-generator) can be very useful here.
2. Go to WC > Orders.
3. Reduce your browser's window size and make sure at least order number, status and date are always shown.
4. Use the devtools on your browser to simulate the page being loaded from a mobile device. I suggest screen widths of 370 - 390px, 400-780px, > 800px. Confirm that order number, status, and date are always shown (even if ellipsized).



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
